netlib/test.py

```diff
@@ -55,22 +56,33 @@ def __init__(self, ssl, q, handler_klass, addr):
                     dhparams, v3_only
         """
         tcp.TCPServer.__init__(self, addr)
-        self.ssl, self.q = ssl, q
+
+        if ssl is True:
+            self.ssl = dict()
+        elif isinstance(ssl, dict):
+            self.ssl = ssl
+        else:
+            self.ssl = None
+
+        self.q = q
         self.handler_klass = handler_klass
         self.last_handler = None
 
     def handle_client_connection(self, request, client_address):
         h = self.handler_klass(request, client_address, self)
         self.last_handler = h
-        if self.ssl:
-            cert = certutils.SSLCert.from_pem(
-                file(self.ssl["cert"], "rb").read()
-            )
-            raw = file(self.ssl["key"], "rb").read()
+        if self.ssl is not None:
+            raw_cert = self.ssl.get(
+                "cert",
+                tutils.test_data.path("data/server.crt"))
+            cert = certutils.SSLCert.from_pem(file(raw_cert, "rb").read())
```

Comment:
We should always use `with open(...)` so that we don't run into issues on pypy (http://pypy.org/compat.html).

(I see this has been there before, but should be fixed nonetheless)